src/components/wallet/methods/trustAsset.ts

```diff
@@ -52,37 +42,33 @@ export default async function trustAsset(
       pincode_stretched
     )
 
-    this.error = null
-    this.loading = { ...this.loading, trust: true }
+    this.logger.instruction(
+      'Loading account to get sequence number for trust transaction'
+    )
+    const account = await this.server.loadAccount(keypair.publicKey())
 
-    await this.server
-      .accounts()
-      .accountId(keypair.publicKey())
-      .call()
-      .then(({ sequence }) => {
-        const account = new Account(keypair.publicKey(), sequence)
-        const transaction = new TransactionBuilder(account, {
-          fee: BASE_FEE,
-          networkPassphrase: Networks.TESTNET,
+    const transaction = new TransactionBuilder(account, {
+      fee: BASE_FEE,
+      networkPassphrase: Networks.TESTNET,
+    })
+      .addOperation(
+        Operation.changeTrust({
+          asset: new Asset(asset, issuer),
         })
-          .addOperation(
-            Operation.changeTrust({
-              asset: new Asset(instructions[0], instructions[1]),
-            })
-          )
-          .setTimeout(0)
-          .build()
+      )
+      .setTimeout(0)
+      .build()
 
-        transaction.sign(keypair)
-        return this.server.submitTransaction(transaction)
-      })
-      .then((res) => console.log(res))
-      .finally(() => {
-        this.loading = { ...this.loading, trust: false }
-        this.updateAccount()
-      })
+    transaction.sign(keypair)
+    this.logger.request('Submitting changeTrust transaction', transaction)
+    const result = await this.server.submitTransaction(transaction)
+    this.logger.response('Submitted changeTrust transaction', result)
+    await this.updateAccount()
```

Comment:
What does updateAccount do? update account balances we have in-memory?

Reply:
yeah it updates balances from horizon, rather than try to make discrete updates based on the action taken